compiler/src/dotty/tools/dotc/transform/FirstTransform.scala

```diff
@@ -188,8 +189,27 @@ class FirstTransform extends MiniPhase with InfoTransformer { thisPhase =>
   override def transformStats(trees: List[Tree])(implicit ctx: Context): List[Tree] =
     ast.Trees.flatten(reorderAndComplete(trees)(ctx.withPhase(thisPhase.next)))
 
-  private def toTypeTree(tree: Tree)(implicit ctx: Context) =
-    TypeTree(tree.tpe).withPos(tree.pos)
+  private object collectBinders extends TreeAccumulator[List[Ident]] {
+    def apply(annots: List[Ident], t: Tree)(implicit ctx: Context): List[Ident] = t match {
+      case t @ Bind(_, body) =>
+        val annot = untpd.Ident(tpnme.BOUNDTYPE_ANNOT).withType(t.symbol.typeRef)
+        apply(annot :: annots, body)
+      case _ =>
+        foldOver(annots, t)
+    }
+  }
+
+  /** Replace type tree `t` of type `T` with `TypeTree(T)`, but make sure all
+   *  binders in `t` are maintained by rewrapping binders around the type tree.
+   *  E.g. if `t` is  `C[t @ (>: L <: H)]`, replace with
+   *  `t @ TC[_ >: L <: H]`. The body of the binder `t` is now wrong, but this does
+   *  not matter, as we only need the info of `t`.
```

Comment:
I understand what the code does but I'm confused by the documentation :). What are `C` and `TC` here? Are you referring to two differents `t` when you say "`t` is  `C[t @ (>: L <: H)]`" ? The comment also says that we rewrap binders around the type tree but the code wraps the type tree with an annotation, not a bind.